official/utils/logging/hooks.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+import examples_per_second_hook
+
+def get_train_hooks(names):
+  """Factory for getting a list of TensorFlow hooks for training by name.
+
+  Args:
+    names: case-insensitive, comma-separated string with names of desired hook
+      classes. Allowed: LoggingTensorHook, ProfilerHook, ExamplesPerSecondHook
+
+  Returns:
+    list of instantiated hooks, ready to be used in a classifier.train call
+
+  Raises:
+    ValueError: if an unrecognized name is passed
+  """
+
+  if not names:
+    raise ValueError('Hook names cannot be None/Empty')
+
+  if not isinstance(names, str):
+    raise ValueError('Hook names should be a string')
+
+  name_list = [name.strip().lower() for name in names.split(',')]
+
+  hooks = []
+  for name in name_list:
+    if name == 'loggingtensorhook':
+      hooks.append(get_logging_tensor_hook())
+    elif name == 'profilerhook':
+      hooks.append(get_profiler_hook())
+    elif name == 'examplespersecondhook':
+      hooks.append(get_examples_per_second_hook())
+    else:
+      raise ValueError('Unrecognized training hook requested: {}'.format(name))
+
+  return hooks
+
+
+def get_logging_tensor_hook():
+  """Returns a LoggingTensorHook with a standard set of tensors that will be
+  printed to stdout.
+  """
+  tensors_to_log = {
+      'learning_rate': 'learning_rate',
+      'cross_entropy': 'cross_entropy',
+      'train_accuracy': 'train_accuracy'
+  }
+
+  return tf.train.LoggingTensorHook(tensors=tensors_to_log, every_n_iter=100)
+
+
+def get_profiler_hook():
+  """Returns a ProfilerHook that writes out timelines that can be loaded into
+  profiling tools like chrome://tracing.
+  """
+  return tf.train.ProfilerHook(save_steps=1000, output_dir='')
+
+
+def get_examples_per_second_hook():
+  return examples_per_second_hook.ExamplesPerSecondHook(batch_size = 256)
```

Comment:
We will have to allow the passing in of args, clearly. Can you modify this module to allow for the passing in of args? A few ways to do this I can think of off the top of my head: have each of these methods take `**kwargs`, and then check for relevant ones (easy to add, but hard to check), or have the factory method above take explicit named args that we add to each time (have to manually add, but guarantee correct signature), or just allow batch_size for now (maybe the best, as if we want to customize everything, we're better off with a config file somewhere). This function should raise an error if there is no batch size, as it's not meaningful to have an arbitrary value substituted in.

Reply:
Yes, agree! For the other two hooks (or maybe others as we add more), I think we also need to allow the users to pass custom args. I am looking into it now. Thank you for the comments!